helix-core/src/main/java/org/apache/helix/model/PauseSignal.java

```diff
@@ -28,7 +28,11 @@
 public class PauseSignal extends HelixProperty {
 
   public enum PauseSignalProperty {
-    REASON
+    REASON,
+    PAUSE_CLUSTER,
```

Comment:
Question: I am not quite sure what do you mean the PAUSE_CLUSTER? Can you elaborate more on each of the entry. FROM_HOST is something I am not clear as well.

Also do we need to explicitly put CANCEL_PENDING_ST or not. We can just make cluster status as PAUSING/EXITING something.

Reply:
These fields are representing the user's action. It'd be a record to help tracking and debugging. 
PAUSE_CLUSTER: whether enable/disable pause mode
FROM_HOST: where the pause signal is sent from, eg. Helix REST or a user's laptop that directly calls java api.
CANCEL_PENDING_ST: whether to cancel pending st. Controller will read this field to determine to send cancel messages.